include/glow/Backends/Backend.h

```diff
@@ -44,21 +53,14 @@ class Backend {
   /// \returns the kind of Backend this is.
   virtual BackendKind getBackendKind() const = 0;
 
-  /// Generate code for input function \param F.
-  virtual std::unique_ptr<CompiledFunction> compile(Function *F) const = 0;
-
-  /// Generate instrumented (for tracing) code for input function \param F.
-  /// When writing a new backend you must override this function to support auto
-  /// instrumentation.
-  virtual std::unique_ptr<CompiledFunction>
-  instrumentAndCompile(Function *F) const {
-    GLOW_UNREACHABLE("Auto instrumentation is not supported by the backend");
-    return compile(F);
+  virtual std::unique_ptr<CompiledFunction> compile(Function *F) const {
+    CompileOptions opts;
+    return compile(F, opts);
   }
 
-  /// Generate code for input function \param F but do not collect constants.
+  /// Generate code for input function \param F.
   virtual std::unique_ptr<CompiledFunction>
-  compileWithoutConstants(Function *F) const = 0;
+  compile(Function *F, CompileOptions &opts) const = 0;
```

Comment:
Should it be `const CompleOptions &opts` or do you intend to change them under certain circumstances?